software/main.py

```diff
@@ -0,0 +1,266 @@
+
+                    robot.move(rotate_dir, rotate_dir, rotate_dir)
+                    #controlls basket colour
+
+                    if basket_color == "magenta":
+                        basket = processedData.basket_m
+                    elif basket_color == "blue":
+                        basket = processedData.basket_b
+
+                    #if that kind of basket is in our list
+                    if basket.exists:
+                        #deltabasket=((basket.x-reso_x_mid)/reso_x_mid)
+                        if(abs(basket.x-reso_x_mid)<10):
+                            robot.stop()
+                            state="throw"
+                            print("Robot is ready to throw")
+                            #state="find_ball"
+                            
+                        elif abs(basket.x-reso_x_mid)>30:
+                            #orbit_speeds = Orbit(radius, reso_x_mid, xcord, dist)
+                             
+                            speedY=0
+                            speedX=0.5
+                            speedR=1000*speedX/(640-radius)
+
+                            if radius > (dist-5):
+                                speedY += (radius-dist)/100
+                            elif radius > (dist+5):
+                                speedY += (radius-dist)/100
+                            
+                            
+                            if xcord > (reso_x_mid + 1) or xcord < (reso_x_mid - 1):
+                                speedR += (reso_x_mid- xcord) / 100 
+                                                   
+                            #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+                            robot.move(speedX, speedR, speedY)                        
+                    else:
+                        #Orbit
+                        #orbit_speeds = Orbit(radius, reso_x_mid, xcord, dist)
+
+                        speedY=0
+                        speedX=0.5
+                        speedR=1000*speedX/(640-radius)
+
+                        if radius > (dist-5):
+                            speedY += (radius-dist)/100
+                        elif radius > (dist+5):
+                            speedY += (radius-dist)/100
+                        
+                        
+                        if xcord > (reso_x_mid + 1) or xcord < (reso_x_mid - 1):
+                            speedR += (reso_x_mid- xcord) / 100 
+                                                
+                        #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+                        robot.move(speedX, speedR, speedY)
+                        
+                        #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+            
+                            
+                    
+
+            elif state == "throw":
+                #calc speed in m/s
+
+                processedData.balls.sort(key= lambda x: x.size)

```

Comment:
Same as before, sort the balls in the image processor class

Reply:
Done